pkg/api/handlers/compat/images_push.go

```diff
@@ -49,9 +49,9 @@ func PushImage(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	authConf, authfile, err := auth.GetCredentials(r)
+	authConf, authfile, key, err := auth.GetCredentials(r)
 	if err != nil {
-		utils.Error(w, "Something went wrong.", http.StatusBadRequest, errors.Wrapf(err, "Failed to parse %q header for %s", auth.XRegistryAuthHeader, r.URL.String()))
+		utils.Error(w, "Something went wrong.", http.StatusBadRequest, errors.Wrapf(err, "Failed to parse %q header for %s", key, r.URL.String()))
```

Comment:
I hate the "Something went wrong" message with a passion.  Can we tell them what went wrong?  Perhaps "Unable to retrieve credentials" or similar?

Reply:
Or I see below you've "Failed to retrieve repository credentials", that works too.

Reply:
@vrothberg 

> @jwhonce, did you check how Atomic Docker deals with the auth header for build?
> 
> We may need a fallback on the server side and check for both headers to be compatible with vanilla Docker (Config) and Atomic (potentially multi-auth Auth).

auth.GetCredentials() pulls whichever header is provided and returns a libpod compatible structure.  So we're covered.